core/domain/exp_services.py

```diff
@@ -169,7 +203,18 @@ def get_exploration_summary_from_model(exp_summary_model):
 
 
 def get_exploration_by_id(exploration_id, strict=True, version=None):
-    """Returns a domain object representing an exploration."""
+    """Returns a domain object representing an exploration.
+
+    Args:
+        exploration_id: str. The id of the Exploration to be returned.
+        strict: bool. When True, discards any invalid exploration id. 
+        version: str or None. The version of the exploration to be returned.
+            If None, the latest version of the exploration is returned.
+
+    Returns:
+        Exploration. The domain object corresponding to the given exploration.        
+    """
```

Comment:
An exception may be raised if strict is True, so there should be a 'Raises' clause.